### PR TITLE
Renaming Content Records to Documents

### LINKED ISSUE #1515: ContentTypeDefinitionRecord and other names are confusing
I think these names are leftovers from O1. Consider renaming them since they are not records anymore.

ContentTypeDefinitionRecord
ContentPartDefinitionRecord
ContentPartFieldDefinitionRecord

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #1515

### CODE REVIEW

**Comment by sebastienros** [comment]
We might still need to do it for Workflows too if that would break.

**Comment by deanmarcussen** [comment]
So this is non breaking for file content definition records (documents), but breaking for the database content definition record I think?

Because there is no type migration support that I know of in YesSql?

It doesn't bother me, can be easily mitigated, with a bit of sql to change the Type name in the Document table.

But of course, there is also at first glance
- Workflow - ActivityRecord
- DataMigrationRecord
- CultureDictionaryRecord (not database related?)

File: src/OrchardCore.Modules/OrchardCore.ContentTypes/RecipeSteps/ContentDefinitionStep.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public Task ExecuteAsync(RecipeExecutionContext context)
             return Task.CompletedTask;
         }
 
-        private void UpdateContentType(ContentTypeDefinition type, ContentTypeDefinitionRecord record)
+        private void UpdateContentType(ContentTypeDefinition type, ContentTypeDefinitionDocument record)
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.ContentTypes/RecipeSteps/ContentDefinitionStep.cs]: Should we rename the variables as well ?

**Comment by sebastienros** [comment]
@deanmarcussen you are totally right about the db Type values, I forgot about it. It should have a custom migration then to do an update statement. Maybe I will wait for a later version then if I can make it completely transparent with migrations.

**Comment by agriffard** [comment]
Many conflicts to resolve.

**Comment by agriffard** [comment]
Still many conflicts to resolve @sebastienros

**Comment by agriffard** [comment]
There is an error executing the unit test `OrchardCore.Tests.Apis.GraphQL.BlogPostTests.WhenThePartHasTheSameNameAsTheContentTypeShouldCollapseFieldsToContentType` during the build :
`Cannot query field \"subtitle\" on type \"BlogPost\`

**Comment by agriffard** [comment]
Readt to be merged.

**Comment by deanmarcussen** [comment]
> @deanmarcussen you are totally right about the db Type values, I forgot about it. It should have a custom migration then to do an update statement. Maybe I will wait for a later version then if I can make it completely transparent with migrations.

@agriffard this is breaking until we do a custom migration for the database content file definition as well

I don't think YesSql has a technique for altering Type values, so it might have to be done as custom sql.

**Comment by hishamco** [comment]
Any update on this?

**Comment by agriffard** [comment]
Closing, many conflicts to fix.
Not sure it will be changed for now.

**Pull request closed without merging**